src/components/CelestialLogo.tsx

```diff
@@ -1,10 +1,29 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import logoImg from '../../public/images/logo.PNG';
 import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
 
 const CelestialLogo = () => {
+  const [scrolling, setScrolling] = useState<Boolean>(false);
+  const router = useRouter();
+
+  useEffect(() => {
+    const handleScroll = () => {
+      setScrolling(window.scrollY > 0);
+    };
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
   return (
-    <Image className='ml-5' width={300} height={100} alt="Logo" src={logoImg} />
+    <div className={`fixed top-0 left-0 z-10 transition-opacity max-sm:w-full ${scrolling ? 'bg-deepblue' : ''}`}>
```

Comment:
What do you think about a short CSS transition on the background? Toggling between two different RGBA colors (one with opacity 0) could be a nice effect 🙂